### PR TITLE
Enable no-invalid-this ESLint rule and pre-apply fixes

### PR DESCRIPTION
**Pull request by tylerbutler**
From [the documentation on the rule](https://typescript-eslint.io/rules/prefer-includes):

> Under the strict mode, `this` keywords outside of classes or class-like objects might be undefined and raise a TypeError.

Disabled for tests and PropertyDDS projects.

### CODE REVIEW

**Comment by Josmithr** [comment]
For the packages in which we're disabling the new rule, should we leave comments about why / future work to re-enable?

**Comment by tylerbutler** [comment]
> For the packages in which we're disabling the new rule, should we leave comments about why / future work to re-enable?

Addressed in a separate PR: https://github.com/microsoft/FluidFramework/pull/10272#discussion_r874139756

File: packages/test/test-version-utils/src/describeCompat.ts [hunk 1]
```diff
@@ -40,8 +40,10 @@ function createCompatSuite(
                         config.containerRuntime,
                         config.dataRuntime,
                     );
+                    // eslint-disable-next-line @typescript-eslint/no-invalid-this
                     Object.defineProperty(this, "__fluidTestProvider", { get: () => provider });
                 });
+                // eslint-disable-next-line @typescript-eslint/no-invalid-this
                 tests.bind(this)((reset: boolean = true) => {
```
Comment (tylerbutler) [inline packages/test/test-version-utils/src/describeCompat.ts:47]: @microsoft/fluid-cr-test Do you know whether these are bugs?

Comment (markfields) [reply packages/test/test-version-utils/src/describeCompat.ts:47]: I have not fully understood how `this` works in all cases yet.  But I know Mocha sets `this` to some context object in some cases/ways, and it appears in the parameter list (but I don't think callers pass it - this is the part I don't totally get). so I think someone could explain why this is correct.  If this were not correct none (ish) of our tests would pass.

**Review by curtisman — APPROVED** [review]

**Pull request merged**